ReflectometryServer/ChannelAccess/reflectometry_driver.py

```diff
@@ -116,7 +116,10 @@ def read(self, reason):
                 elif is_pv_name_this_field(SERVER_MESSAGE, reason):
                     return STATUS_MANAGER.message
                 elif is_pv_name_this_field(SERVER_ERROR_LOG, reason):
-                    return STATUS_MANAGER.error_log
+                    # The server status manager class appends new messages to the end of the log string,
+                    # so the last "count" characters are returned.
+                    error_log_max_character_size = self._pv_manager.PVDB[SERVER_ERROR_LOG]["count"]
+                    return STATUS_MANAGER.error_log[-error_log_max_character_size:]
```

Comment:
We should be able to write tests for this at the IOC test level.

- Make the reflectometry server log more than 10k characters
- Use the existing `assert_log_messages` context manager in the tests to assert that *all* messages end up in the log file (the context manager can return a reference to all messages produced if you need to do complex assertions)
- Verify that the error log served over CA is <= 10k characters and has been truncated